### ISSUE TITLE
Add support the coordiator job submitted to run in far future 

### ISSUE THREAD

**Issue by angeloh** [YahooArchive/oozie#46]
The following code in PriorityDelayQueue.java class

```
public int compareTo(Delayed o) {
        return (int) (getDelay(TimeUnit.MILLISECONDS) - o.getDelay(TimeUnit.MILLISECONDS));
}
```

should be replace by something like this:

```
public int compareTo(Delayed o) {
        long diff = (getDelay(TimeUnit.MILLISECONDS) - o.getDelay(TimeUnit.MILLISECONDS));
        if(diff > 0) {
            return 1;
         } else if(diff < 0) {
            return -1;
         } else {
            return 0;
         } 
}
```

**Comment by angeloh** [comment]
XLogStreamer.FileInfo has a compareTo(). This should be taken care too.

**Comment by tucu00** [comment]
Angelo,

Would you create a topical branch GH-0048 in your own repo with the patch?

Thanks.

Alejandro

**Comment by rvs** [comment]
This has now been incorporated into the proposed maintenance branch 2.2. We will be sending a dedicated pull request to review commits from there when they go to master. That said, please DO  leave your feedback rvs/oozie@9453b4cc2377dbab56144e2b2dcebb9e83f78a98

**Comment by angeloh** [comment]
Closed by 3f75f8168acbc7fbd4d1187efe0e7f496e6a2bcd Add support the coordiator job submitted to run in far future
